views/campgrounds/index.ejs

```diff
@@ -1,159 +1,166 @@
-            <button type="submit" class="btn btn-primary" id="sort-btn" style="margin-left:-5px">Sort</button>
-        </form>
-    </div>
-    <!-- Search Bar (right-aligned) -->
-    <form action="/campgrounds/search" method="GET" class="form-inline my-2 my-lg-0" id="search-form">
-        <div class="input-group">
-            <input class="form-control" type="search" style=" border:1px solid rgb(35, 35, 35)" placeholder="Search for campsites..." aria-label="Search" name="q" id="search-input">
-            <div class="input-group-append">
-                <button class="btn btn-info" type="button" id="search-button">Search</button>
+    <div id="cluster-map"></div>
+    <h1>All Campgrounds</h1>
+
+    <link rel="stylesheet" href="stylesheets/searchCampground.css">
+
+    <div class="campground-search-controls">
+        <!-- Search Form -->
+        <form action="/campgrounds/search" method="GET" class="campground-search-form" id="search-form">
+            <div class="campground-input-group">
+                <input class="campground-search-input" type="search" placeholder="Search for campsites..."
+                    aria-label="Search" name="q" id="search-input">
+                <button class="campground-search-button" type="submit" id="search-button">Search</button>
             </div>
+        </form>
+
+        <!-- Sort Controls -->
+        <div class="campground-sort-controls">
+            <button id="show-nearest-btn" class="campground-nearest-btn">Show Nearest to Me</button>
+            <form action="/campgrounds" method="GET" class="d-inline">
+                <select name="sort" id="sort" class="campground-sort-select">
+                    <option value="">--Select--</option>
+                    <option value="priceAsc" <%=(locals.query && query.sort==='priceAsc' ) ? 'selected' : '' %>>Price:
+                        Low
+                        to High</option>
+                    <option value="priceDesc" <%=(locals.query && query.sort==='priceDesc' ) ? 'selected' : '' %>>Price:
+                        High to Low</option>
+                    <option value="reviewsHighest" <%=(locals.query && query.sort==='reviewsHighest' ) ? 'selected' : ''
+                        %>
+                        >Reviews: Highest Rated</option>
+                    <option value="reviewsLowest" <%=(locals.query && query.sort==='reviewsLowest' ) ? 'selected' : ''
+                        %>
+                        >Reviews: Lowest Rated</option>
+                </select>
+                <button type="submit" class="campground-sort-button">Sort</button>
+            </form>
         </div>
-    </form>
-</div>
-
-<!-- Campgrounds List -->
-<div id="campground-container">
-    <% for(let campground of campgrounds){ %>
-        <div class="card text-white bg-dark mb-3" id="campinfo">
-            <div class="row">
-                <div class="col-md-4" id="campgroundhomeimage">
+    </div>
+
+    <div class="campground-listing-grid">
```

Comment:
please add this id here ``id="campground-container"``. search functionality is not working because of that